### ISSUE TITLE
Systray icon is missing

### ISSUE THREAD

**Issue by rpallai** [flathub/com.viber.Viber#4]
There is no systray icon in Flathub build. Which means it is not possible to quit the application because closing of the main window just send it to the background and no way to bring it back.

**Comment by TingPing** [comment]
Hmm, permissions for it are added, what desktop and can you confirm it works on a non-flatpak install?

**Comment by rpallai** [comment]
I'm on Plasma 5 (Fedora 26). Yes, it works on a native install, the systray icon always visible.

**Comment by sagarchalise** [comment]
This happens on gnome-shell 3.26 with "kstatusnotifier/topiconsplus extension" as well. 

`failed to register service "org.kde.StatusNotifierItem-3-2` is output when run from shell.

**Comment by antonb-perfecto** [comment]
Confirmed on Linux Manjaro KDE5. Native viber app has icon, flatpak one doesn't.

**Comment by kerozoli** [comment]
Confirmed on SuSe Leap 15.0 KDE 5.12

**Comment by TingPing** [comment]
Try `flatpak run --own-name='org.kde.*' com.viber.Viber`

**Comment by summoner001** [comment]
Confirmed!

**Comment by rpallai** [comment]
@TingPing It works with this parameter.

**Comment by eliaskousk** [comment]
@TingPing Thanks, adding the `--own-name='org.kde.*'` parameter on my existing menu launcher parameters for Viber made the icon appear like before (without FlatPak). I'm using Kubuntu 18.04.

**Comment by wojciech-kopras** [comment]
@TingPing Thanks, it works.

@summoner001 To make it working without modyfing launchers I did:

`flatpak override --user --own-name='org.kde.*' com.viber.Viber`

You can do so with or without this `--user` switch. But of course it would be nice to have this bug corrected -- I don't know flatpak that much, I don't know the other implications of adding `--own-name='org.kde.*'`..

**Comment by summoner001** [comment]
@wojciech-kopras Thank You! It works for now.

**Comment by Xtigyro** [comment]
Confirm on Kubuntu 18.04. Also - the workaround fixes the bug.

**Comment by darkdragon-001** [comment]
Probably, `--own-name=` and `--talk-name=` need to be added to the manifest as done [here](https://github.com/flathub/com.github.bajoja.indicator-kdeconnect/blob/master/com.github.bajoja.indicator-kdeconnect.json#L16-L22).

**Comment by zaro** [comment]
Can confirm the same with Fedora 32 and Gnome. The woraround solves it.

**Comment by mitsos1os** [comment]
Problem confirmed in PopOS! 20.10 as well.
The proposed solution from @wojciech-kopras works!
Thanks!

**Comment by rajeshisnepali** [comment]
> `flatpak override --user --own-name='org.kde.*' com.viber.Viber`

Isn't it possible to set the above command by default in `/var/lib/flatpak/exports/share/applications/com.viber.Viber.desktop` everytime the user install viber? 

I mean the developers could fix it, right?

**Comment by 7system7** [comment]
From GNOME 40, it is not working anymore. The viber icon looks like this:

![Screenshot from 2021-11-07 19-59-27](https://user-images.githubusercontent.com/3964568/140658212-471e56f8-fddd-4d3b-be6e-2876ffeab60d.png)

**Comment by akurtakov** [comment]
Do you still see the issue after latest updates?

**Comment by upanasiuk** [comment]
> From GNOME 40, it is not working anymore. The viber icon looks like this:
> 
> ![Screenshot from 2021-11-07 19-59-27](https://user-images.githubusercontent.com/3964568/140658212-471e56f8-fddd-4d3b-be6e-2876ffeab60d.png)

Try `flatpak run --own-name='org.kde.*' --env=XDG_CURRENT_DESKTOP='GNOME' com.viber.Viber`

**Comment by zaro** [comment]
Just tested by installing from Flathub on Fedora 35 (Gnome 41), and there is no icon. If I start it with the command suggested by @upanasiuk  the icon is shown normally.

**Comment by trancexpress** [comment]
> > From GNOME 40, it is not working anymore. The viber icon looks like this:
> > ![Screenshot from 2021-11-07 19-59-27](https://user-images.githubusercontent.com/3964568/140658212-471e56f8-fddd-4d3b-be6e-2876ffeab60d.png)
>
> Try `flatpak run --own-name='org.kde.*' --env=XDG_CURRENT_DESKTOP='GNOME' com.viber.Viber`

I did some updates today (quite a few so I'm not sure which package was the problem), apparently I now also have to specify the ENV variable. I'm on KDE Plasma.

Prior to the updates it was enough to run this:

```
flatpak run --own-name='org.kde.*' com.viber.Viber
```

After the updates the tray icon would be "invisible", it took space in the tray bar but was not actually shown. The badge for notifications (why I want the tray icon in the first place) was also not shown.

The tray icon (with working badges) is shown if I run with:

```
flatpak run --own-name='org.kde.*' --env=XDG_CURRENT_DESKTOP='KDE' com.viber.Viber
```

**Comment by nixonwhite** [comment]
On Gnome 42 there is also tray icon doesn't shows

The tray icon is shown if run:
`flatpak run --own-name='org.kde.*' --env=XDG_CURRENT_DESKTOP='KDE' com.viber.Viber`

**Comment by Tamas-Toth-ebola** [comment]
I can also confirm that on Ubuntu 22.04 with 'its' GNOME 'version' originally there is olny just 3 dots as icon but...

`flatpak run --own-name='org.kde.*' --env=XDG_CURRENT_DESKTOP='GNOME' com.viber.Viber`

...correct i!

**Comment by akurtakov** [comment]
In latest build I've switched the flatpak runtime to Gnome for other reasons. Would you please check whether if fixes this one too?

**Comment by Tamas-Toth-ebola** [comment]
After your https://github.com/flathub/com.viber.Viber/issues/49 fix, and a secondary update today Viber is starting fine, but sadly without any type of 'tray' icon. There is no good or wrong icon, there is simply no icon.

_(On 'stock' Ubuntu 22.04 with its GNOME)_

**Comment by akurtakov** [comment]
Do you get tray icons from other flatpak applications? I would have to look at how someone else achieves it.

**Comment by nixonwhite** [comment]
Flatpak'ed version of Telegram app has tray icon and it works.

**Comment by Tamas-Toth-ebola** [comment]
> Do you get tray icons from other flatpak applications? I would have to look at how someone else achieves it.

Sadly currently I do not use any other. But before the latest update therefore the current 'release' the earlier worked fine with the `flatpak run --env=XDG_CURRENT_DESKTOP='GNOME' com.viber.Viber` help.

**Comment by streger** [comment]
I'm having the same issue on Ubuntu 22.04 Mate, didn't had issue on older Ubuntu Mate releases (21.10, 21.04, 20.10, 20.04).
The issue is also present with .deb Viber installation now.
Currently have the  `/usr/bin/flatpak run --branch=stable --arch=x86_64 --env=XDG_CURRENT_DESKTOP='MATE' --command=viber com.viber.Viber` but it doesn't help.

**Comment by gobalius** [comment]
I have the same issue - no Viber icon in system tray on CentOS 7.9 with GNOME 3.28.3. None of the above mentioned workarounds work for me. The latest RPM version of Viber that I can run (with manually adding missing libraries to the binary folder) is 13.3.1 and I would just be using that one if it wasn't so unstable and if it wasn't missing new Viber functionality ("liking" messages with the hear icon).
I'm using the Flathub Viber version 18.2.0.2

**Comment by streger** [comment]
Using new Viber 18.2.0.2 does show an icon in Ubuntu Mate 22.04. "Indicator Applet Complete" aka panel "tray" both with Flatpak (without proper icon, a default red missing icon) and with .deb installation (proper Viber logo icon), but in both of those cases that icon doesn't do anything. Clicking on it while the Viber window was closed just opens an ultra small menu placeholder with nothing in it, and I still have to force kill Viber to exit it and open again to get back main window.
This all leads me to believe that it's actually a Viber problem.

**Comment by Tamas-Toth-ebola** [comment]
@streger I'm a bit confused about your last 2 comments. On 08.12 when you wrote that you had no tray icon, Viber 18.2.0.2 had been already available. I wrote my comment based on this version. Which version did you use then? Just because in the last 12 days there is no new Viber version, and no new Flatpak installer for it, so I'm a bit confused how you got 2 different results?!

Do I misunderstand anything?

**Comment by streger** [comment]
@Tamas-Toth-ebola As you've noticed in my first comment unfortunately I haven't noted the Viber version, which was definitely a  mistake on my part. The version I've described there, in both .deb and Flatpak version were same and it was just version behind this last Viber. The dates don't matter, since at the time of my first comment I didn't had the latest version. I've installed latest version 18.2.0.2 just last week.

**Comment by gobalius** [comment]
I can find in my /var/log/messages these entries:
`Aug 24 13:30:45 my-pc-name journal: [AppIndicatorSupport-FATAL] invalid image format: /home/my-username/.var/app/com.viber.Viber/cache/runtime-my-username/qt-trayicon-ZwnrsD.png
Aug 24 13:30:45 my-pc-name journal: [AppIndicatorSupport-FATAL] invalid image format: /home/my-username/.var/app/com.viber.Viber/cache/runtime-my-username/qt-trayicon-ZwnrsD.png
Aug 24 13:30:45 my-pc-name journal: [AppIndicatorSupport-FATAL] invalid image format: /home/my-username/.var/app/com.viber.Viber/cache/runtime-my-username/qt-trayicon-ZwnrsD.png
Aug 24 13:30:45 my-pc-name journal: [AppIndicatorSupport-FATAL] invalid image format: /home/my-username/.var/app/com.viber.Viber/cache/runtime-my-username/qt-trayicon-ZwnrsD.png
Aug 24 13:30:45 my-pc-name journal: [AppIndicatorSupport-FATAL] invalid image format: /home/my-username/.var/app/com.viber.Viber/cache/runtime-my-username/qt-trayicon-ZwnrsD.png
Aug 24 13:30:45 my-pc-name journal: [AppIndicatorSupport-FATAL] invalid image format: /home/my-username/.var/app/com.viber.Viber/cache/runtime-my-username/qt-trayicon-ZwnrsD.png
Aug 24 13:34:26 my-pc-name journal: [AppIndicatorSupport-FATAL] invalid image format: /home/my-username/.var/app/com.viber.Viber/cache/runtime-my-username/qt-trayicon-QzLLOE.png
Aug 24 13:34:26 my-pc-name journal: [AppIndicatorSupport-FATAL] invalid image format: /home/my-username/.var/app/com.viber.Viber/cache/runtime-my-username/qt-trayicon-QzLLOE.png
Aug 24 13:34:26 my-pc-name journal: [AppIndicatorSupport-FATAL] invalid image format: /home/my-username/.var/app/com.viber.Viber/cache/runtime-my-username/qt-trayicon-QzLLOE.png
Aug 24 13:34:26 my-pc-name journal: [AppIndicatorSupport-FATAL] invalid image format: /home/my-username/.var/app/com.viber.Viber/cache/runtime-my-username/qt-trayicon-QzLLOE.png
Aug 24 13:34:26 my-pc-name journal: [AppIndicatorSupport-FATAL] invalid image format: /home/my-username/.var/app/com.viber.Viber/cache/runtime-my-username/qt-trayicon-QzLLOE.png
Aug 24 13:34:26 my-pc-name journal: [AppIndicatorSupport-FATAL] invalid image format: /home/my-username/.var/app/com.viber.Viber/cache/runtime-my-username/qt-trayicon-QzLLOE.png`

If I try to open these files with The GNOME image viewer, they are shown normally...

**Comment by gobalius** [comment]
OK, I have resolved my issue... I switched from the GNOME extension "AppIndicator and KStatusNotifierItem Support" to "Tray Icons". Go figure.

**Comment by vig0gne** [comment]
In some cases (in my example is Ubuntu 22.04 with GNOME with Dash to Panel plugin) helps [this](https://askubuntu.com/questions/774620/viber-indicator-not-working?rq=1) issue:
 change **Exec=/opt/viber/Viber %u** string in _viber.desktop_ file to **Exec=dbus-launch /opt/viber/Viber %u**

**Comment by glnemeth** [comment]
> In some cases (in my example is Ubuntu 22.04 with GNOME with Dash to Panel plugin) helps [this](https://askubuntu.com/questions/774620/viber-indicator-not-working?rq=1) issue: change **Exec=/opt/viber/Viber %u** string in _viber.desktop_ file to **Exec=dbus-launch /opt/viber/Viber %u**

Thank you very much! I was struggling with that for months now and that fixed it, finally I can see if I have new messages!
Just a note, I had to remove the snap/flatpack version of Viber and install the download with dpkg -i, only after that did I have the /usr/share/applications/viber.desktop file, which I could change like mentioned above.

**Comment by Tamas-Toth-ebola** [comment]
Hi!

Just would like to report, that after upraded to Ubuntu 22.10 'beta', the tray icon have been appeared but without right click context menu. So now I can see the running application but still can not exit it from its tray icon menu.

Viber version is the latest flatpak version: 18.2.0.2

App start had made by the default Flatpak launcher at `/var/lib/flatpak/app/com.viber.Viber/current/.../export/share/applications/com.viber.Viber.desktop`
```
Exec=/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=viber com.viber.Viber
```

**Comment by streger** [comment]
Hi again,

Just to confirm that same behavior is happening to me as for @Tamas-Toth-ebola but with .deb Viber as well on Ubuntu Mate 22.04.
Interestingly, I have another Ubuntu Mate 22.04 device and there the tray-icon right-click context menu work as expected, it pops-up populated with items, of which one is to exit (quit).
So, now, I'm suspecting this is a graphic card thing, since my first device, where it's not working is an AMD CPU with dedicated nVidia graphic card. The second device, where it's working good, is an Intel CPU with no dedicated graphic card.

**Comment by Tamas-Toth-ebola** [comment]
Hi @streger , what you wrote is more than strange for me, but I can confirm that my device is an Intel NUC8i7, of course with all integrated Intel solutions _(CPU/IGP, etc)_.

**Comment by streger** [comment]
Well, that's even stranger, since on your Intel device it's not working. And there I was thinking that's the issue.
One of the reasons why I thought that might be the issue, is because I've compared complete dconf settings between my two devices and haven't found anything different that might be the cause for the issue.
It seems that I'll have to do a clean install of either this stable or when the 22.10 comes out on the device where's it's broken (it was an upgrade from previous Ubuntu)
Thanks for the update @Tamas-Toth-ebola.

**Comment by Alt37** [comment]
> the tray icon had been appeared but without right click context menu. So now I can see the running application but still can not exit it from its tray icon menu.
I can confirm that with Ubuntu 22.10 (or [gnome-shell-extension-appindicator v46](https://github.com/ubuntu/gnome-shell-extension-appindicator) tray icon shows up but doesn't react to right click.
Right click menu shows up on KDE Plasma though (tested v5.24.7).

**Comment by Tamas-Toth-ebola** [comment]
I can confirm what @Alt37 mentioned. On 'stock' _(from this aspect)_ Ubuntu 22.10 nightly, after installing `dbus-x11` package, despite I'm on Wayland the solution works.

So summarized: after installing `dbus-x11`, and edit the corresponding flatpak app shortcut _(`/var/lib/flatpak/app/com.viber.Viber/current/active/export/share/applications/com.viber.Viber.desktop`)_ to `Exec=dbus-launch /usr/bin/flatpak run --branch=stable --arch=x86_64 --command=viber com.viber.Viber` instead of original execution attribution, the context menu of the systray icon works.

But there are some comments on it, just to know:

- The menu is not the same type of menu which 'every other' normal systray icons offer, but at least it works.
- The very first right-click not works, we have to click on it more times.
- If I'm right officially currently GNOME Panel do not differentiate left- or right-click on them. As the menu is not the same as the others, left-click open the app, while right-click open the contex menu.
- With opened menu, left-click outside of it not close the menu. We have to explicitly right-click on it again, or choose one of the menu items.

**Comment by akurtakov** [comment]
There is new version of Viber on latest 23.08 freedesktop runtime. Please update and let us know if it still fails for you?

**Comment by Tamas-Toth-ebola** [comment]
The current Flathub version is theoretically version 20.3.0.1 _(https://flathub.org/apps/com.viber.Viber)_, but in practice version 21.0.0.1 works fine for me, but basically already worked fine in the last months, so thanks for it.
